src/instructions/common.rs

```diff
@@ -85,7 +85,8 @@ fn check_load_boundary<R: Register>(version0: bool, address: &R, bytes: u64) ->
     if version0 {
         let address = address.to_u64();
         let end = address.checked_add(bytes).ok_or(Error::MemOutOfBound)?;
-        if end == RISCV_MAX_MEMORY as u64 {
+        // The memory size of CKB-VM V0 is 4M.
+        if end == RISCV_MAX_MEMORY_V0 as u64 {
```

Comment:
Do we need to manually pass-in the memory size?

Reply:
I still have a question here: should this function accept memory size as a parameter, and use real memory size to do the checking?

Reply:
It's a good proposal, we should do it